docs/source/app_developers_guide/testing.rst

```diff
@@ -0,0 +1,145 @@
+*******************************
+Testing a Transaction Processor
+*******************************
+
+Say we've written a transaction processor -- how can it be tested? In this section, we'll see how to test the XO transaction processor described in :doc:`txn_family_tutorial` against a mock validator.
+
+Communication with the Validator
+================================
+
+First, let's look at what happens when the XO transaction 'create game000' is processed. Communication between the validator and a transaction processor (TP) takes place in six steps:
+
+1) The validator needs to process a transaction, so it sends a transaction process request to the TP containing the transaction information (in this case, the command 'create' and the game name 'game000'.
+
+2) The TP needs to know whether there already exists a game with the name 'game000', so it sends a 'get' request back to the validator.
+
+3) After querying state, the validator sends back the results to the TP (in this case, we'll assume that there is no game called 'game000').
+
+4) The TP now sends a message back to the validator requesting that it set 'game000' to an initialized game (with an empty board, etc).
+
+5) The validator sends the address of the created game back to the TP.
+
+6) The TP sends back an 'OK' response.
+
+This description suggests an outline for a test:
+
+.. code-block:: python
+
+    def test_create_game(self):
+        self.validator_sends_tp_process_request(
+            action='create',
+            game='game000')
+
+        get_message = self.validator_expects_get_request('game000')
+        
+        self.validator_responds_to_get_request(
+            get_message,
+            game='game000', board=None,
+            state=None, player1=None, player2=None)
+
+        set_message = self.validator_expects_set_request(
+            game='game000', board='---------',
+            state='P1-NEXT', player1='', player2='')
+        
+        self.validator_responds_to_set_request(set_message, 'game000')
+        
+        self.validator_expects_tp_response('OK')
+
+Message Factory and Mock Validator
+==================================
+
+To implement these functions, we'll need a mock validator to send messages to and from and a way manufacture the appropriate messages. Manufacturing messages is not terribly interesting -- it's just a matter of getting the relevant information into probuf objects. This will be taken care of for us by the *XoMessageFactory* (``/project/sawtooth-core/sdk/examples/xo_python/sawtooth_xo/xo_message_factory.py``), which itself relies on the more general *MessageFactory* (``/project/sawtooth-core/sdk/python/sawtooth_processor_test/message_factory.py``).
```

Comment:
Actually I would say that the protobuf stuff is very involved, which is part of the reason I find it so uninteresting! But I take your point, and I'll make it more neutral.